core/server/api/endpoints/utils/serializers/output/utils/extra-attrs.js

```diff
@@ -3,30 +3,59 @@ const readingMinutes = require('@tryghost/helpers').utils.readingMinutes;
 module.exports.forPost = (frame, model, attrs) => {
     const _ = require('lodash');
 
-    if (!Object.prototype.hasOwnProperty.call(frame.options, 'columns') ||
-        (frame.options.columns.includes('excerpt') && frame.options.formats && frame.options.formats.includes('plaintext'))) {
-        if (_.isEmpty(attrs.custom_excerpt)) {
-            let plaintext = model.get('plaintext');
-
-            if (plaintext) {
-                attrs.excerpt = plaintext.substring(0, 500);
+    // This function is split up in 3 conditions for 3 different purposes:
+    // 1. Gets excerpt from post's plaintext. If custom_excerpt exists, it overrides the excerpt but the key remains excerpt.
+    if (Object.prototype.hasOwnProperty.call(frame.options, 'columns') || _.includes(frame.options.columns, 'excerpt') || _.includes(frame.options.columns, 'excerpt') && frame.options.formats && frame.options.formats.includes('plaintext')) {
+        if (_.includes(frame.options.columns, 'excerpt')) {
+            if (!attrs.custom_excerpt || attrs.custom_excerpt === null) {
+                let plaintext = model.get('plaintext');              
```

Comment:
fyi, there's loads of whitespaces in the end of this line. we don't want any of that :upside_down_face:

Reply:
Fixed! :)